From d4442c3df7552756f53e656e446bc1bd7dc79a88 Mon Sep 17 00:00:00 2001
From: Fabrice Fontaine <fontaine.fabrice@gmail.com>
Date: Mon, 18 Jun 2018 19:52:58 +0200
Subject: [PATCH] websockets: _GNU_SOURCE needed for S_IF{DIR,REG}

Define of _GNU_SOURCE is needed to be able to use S_IFDIR and S_IFREG in
src/websockets.c

Fixes:
 - http://autobuild.buildroot.net/results/7dcfb6ca9d14a5cd6872590065549356f1ab42a0

[Upstream status: https://github.com/eclipse/mosquitto/pull/862]

Signed-off-by: Fabrice Fontaine <fontaine.fabrice@gmail.com>
---
 src/websockets.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/src/websockets.c b/src/websockets.c
index 1e513ae..7722b4d 100644
--- a/src/websockets.c
+++ b/src/websockets.c
@@ -29,6 +29,8 @@ POSSIBILITY OF SUCH DAMAGE.
 
 #ifdef WITH_WEBSOCKETS
 
+#define _GNU_SOURCE
+
 #include "config.h"
 
 #include <libwebsockets.h>
-- 
2.14.1

